src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,333 @@
+import { useNavigate } from "react-router-dom";
+import { queryKeys } from "util/queryKeys";
+import { createInstance } from "api/instances";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { useSettings } from "context/useSettings";
+import InstanceLink from "../InstanceLink";
+import {
+  UploadExternalFormatFileFormValues,
+  uploadExternalFormatFilePayload,
+  FILE_CHUNK_SIZE,
+  isImageTypeRaw,
+  sendFileByWebSocket,
+  supportedVMArchOptions,
+} from "util/uploadExternalFormatFile";
+import { getInstanceName } from "util/operations";
+import { getFileExtension } from "util/helpers";
+import ScrollableContainer from "components/ScrollableContainer";
+import classnames from "classnames";
+
+interface Props {
+  close: () => void;
+  uploadState: UploadState | null;
+  setUploadState: (state: UploadState | null) => void;
+  fixedFormHeight: boolean;
+}
+
+const UploadExternalFormatFileForm: FC<Props> = ({
+  close,
+  uploadState,
+  setUploadState,
+  fixedFormHeight,
+}) => {
+  const { project, isLoading } = useProject();
+  const instanceNameAbort = useState<AbortController | null>(null);
+  const toastNotify = useToastNotification();
+  const navigate = useNavigate();
+  const eventQueue = useEventQueue();
+  const queryClient = useQueryClient();
+  const { data: settings } = useSettings();
+  const [isRawImage, setIsRawImage] = useState(false);
+
+  const handleUploadProgress = (
+    instanceName: string,
+    current: number,
+    total: number,
+  ) => {
+    setUploadState({
+      percentage: Math.floor((current / total) * 100),
+      loaded: current,
+      total: total,
+    });
+
+    if (current === total) {
+      close();
+      toastNotify.info(
+        <>
+          Upload completed. Now creating instance{" "}
+          <strong>{instanceName}</strong>.
+        </>,
+      );
+      navigate(`/ui/project/${project?.name}/instances`);
+    }
+  };
+
+  const handleUploadError = (error: Error) => {
+    toastNotify.failure("Image upload failed.", error);
+  };
+
+  const handleSuccess = (instanceName: string) => {
+    const message = (
+      <>
+        Created instance{" "}
+        <InstanceLink
+          instance={{ project: project?.name || "", name: instanceName }}
+        />
+        .
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () =>
+          navigate(
+            `/ui/project/${project?.name}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const handleFailure = (msg: string) => {
+    toastNotify.failure("Instance creation failed.", new Error(msg));
+  };
+
+  const invalidateCache = () => {
+    void queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.instances;
+      },
+    });
+  };
+
+  const handleConfirmUpload = (values: UploadExternalFormatFileFormValues) => {
```

Comment:
```suggestion
  const handleSubmit = (values: UploadExternalFormatFileFormValues) => {
```
There is no confirm here, is there?